modules/client/src/main/java/org/jboss/wsf/stack/cxf/client/ProviderImpl.java

```diff
@@ -564,27 +568,51 @@ public void publish(jakarta.xml.ws.spi.http.HttpContext context)
    /**
     * An extension of the org.apache.cxf.jaxws.ServiceImpl allowing for
     * setting JBossWS client default config handlers.
-    *
     */
    static final class JBossWSServiceImpl extends ServiceImpl {
 
+      private final ConcurrentMap<Integer, SoftReference<Object>> portsCache = new ConcurrentHashMap<>();
+
+      private static <T> int keyOf(final QName portName, final EndpointReferenceType epr, final Class<T> serviceEndpointInterface, final WebServiceFeature... features) {
+         int result = 17;
+         result = 37 * result + (portName == null ? 0 : portName.hashCode());
+         result = 37 * result + (epr == null ? 0 : epr.hashCode());
+         result = 37 * result + (serviceEndpointInterface == null ? 0 : serviceEndpointInterface.hashCode());
+         result = 37 * result + Arrays.hashCode(features);
+         return result;
+      }
+
```

Comment:
can we directly use Objects.hash(portName, epr, serviceEndpointInterface, Arrays.hashCode(features)) here?

Reply:
Sure, I wasn't aware about such utility class.